### PR TITLE
Add Header Modification Rules

### PR DESCRIPTION
**Pull request by pedrosousa**
Addresses https://jira.cfops.it/browse/PCX-1237.

### CODE REVIEW

File: products/rules/src/content/transform/create-header-modification-rule.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+</Aside>
+
+To create a new HTTP Header Modification Rule:
+ 
+1. Log in to the Cloudflare dashboard.
+
+1. Select the **Websites** tab and choose the site for which you want to create a new HTTP Header Modification Rule.
+
+1. Click the **Rules** tile in the app bar, and then click the **Transform Rules** tab.
```
Comment (kodster28) [inline products/rules/src/content/transform/create-header-modification-rule.md]: Know we haven't standardized this, but is it simpler to say **Rules** > **Transform Rules**?

File: products/rules/src/content/transform/create-header-modification-rule.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+1. Under **When incoming requests match**, define the rule expression using either the Expression Builder inputs and dropdowns or the Expression Editor. For more information on Expression Builder and Expression Editor, see [Expression Builder and Editor](https://developers.cloudflare.com/firewall/cf-dashboard/create-edit-delete-rules#expression-builder-and-editor) in the Firewall documentation.
+
+    <Aside type='note'>
+
+    Check the [available fields and functions](#available-fields-and-functions-for-setting-http-header-values) during the Beta.
+
+    </Aside>
+
+1. Select an action for your Header Modification Rule from the dropdown under **Modify header**. Select one of the following options:
```
Comment (kodster28) [inline products/rules/src/content/transform/create-header-modification-rule.md]: Would this be simpler as, "For **Modify header**, select one of the following options:"

File: products/rules/src/content/transform/create-header-modification-rule.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    ![Transform Rules tab](../static/transform/overview.png)
+
+1. Click **Create transform rule** > **Modify Header**.
+
+    ![Create HTTP Header Modification rule page](../static/transform/create-header-modification-rule.png)
+ 
+1. In the page that displays, enter a descriptive name for the URL Rewrite Rule in **Rule name**.
+
+1. Under **When incoming requests match**, define the rule expression using either the Expression Builder inputs and dropdowns or the Expression Editor. For more information on Expression Builder and Expression Editor, see [Expression Builder and Editor](https://developers.cloudflare.com/firewall/cf-dashboard/create-edit-delete-rules#expression-builder-and-editor) in the Firewall documentation.
```
Comment (kodster28) [inline products/rules/src/content/transform/create-header-modification-rule.md]: Could you just shorten to "define the [rule expression](https://developers.cloudflare.com/firewall/cf-dashboard/create-edit-delete-rules#expression-builder-and-editor)."?

File: products/rules/src/content/transform/create-header-modification-rule.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    <Aside type='note'>
+
+    Check the [available fields and functions](#available-fields-and-functions-for-setting-http-header-values) during the Beta.
+
+    </Aside>
+
+1. Select an action for your Header Modification Rule from the dropdown under **Modify header**. Select one of the following options:
+ 
+    * _Set static_ — Sets the value of an HTTP header to a static string value. Overrides the value of an existing HTTP header with the same name, or adds a new HTTP header if it does not exist.
```
Comment (kodster28) [inline products/rules/src/content/transform/create-header-modification-rule.md]: Remove the comma after "name"

File: products/rules/src/content/transform/create-url-rewrite-rule.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+# Create a URL Rewrite Rule
+
+To create a new URL Rewrite Rule:
+
+1. Log in to the Cloudflare dashboard.
+
+1. Select the **Websites** tab and choose the site for which you want to create a new URL Rewrite Rule.
+
+1. Click the **Rules** tile in the app bar, and then click the **Transform Rules** tab.
```
Comment (kodster28) [inline products/rules/src/content/transform/create-url-rewrite-rule.md]: Same comment on the navigation directions as previously.

File: products/rules/src/content/transform/create-url-rewrite-rule.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    ![Create Transform Rule page](../static/transform/create-url-rewrite-rule.png)
+
+1. In the page that displays, enter a descriptive name for the rule in **Rule name**. The rule name in this example is _Welcome GB Users_.
+
+1. Under **When incoming requests match**, use the **Field** drop-down list to choose an HTTP property. For each request, the value of the property you choose for **Field** is compared to the value you specify for **Value**.
+
+    ![Selecting a field in the Expression Builder](../static/transform/expression-field.png)
+
+1. Use the **Operator** drop-down list to choose a comparison operator. For an expression to match, the value of the request **Field** and the value specified in the **Value** input must satisfy the comparison operator.
```
Comment (kodster28) [inline products/rules/src/content/transform/create-url-rewrite-rule.md]: If possible, could you reduce the words in this step? Thinking you don't need to have the bit about "text-only version" b/c it's called out in the step below.

File: products/rules/src/content/transform/create-url-rewrite-rule.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+1. Specify the value to match. If the value is an enumeration, then the **Value** control will be a drop-down list. Otherwise, it will be a text input. In this example the value _United Kingdom_ is set using the **Country** drop-down list.
+
+    ![Entering a field value in the Expression Builder](../static/transform/expression-value.png)
+
+1. Add any other filters to the rule expression. The example defines an additional filter for the _URI Path_, which must be equal to `/welcome.html`.
+
+1. To define the action for your URL Rewrite Rule, select one of the available options displayed as radio buttons, and then a value from the drop-down list, depending on the action.
+
+    In this example, the path rewrite _Static_ action tells Cloudflare to rewrite the original path changing it to the provided path, `/welcome-gb.html`.
```
Comment (kodster28) [inline products/rules/src/content/transform/create-url-rewrite-rule.md]: Maybe switch to "to rewrite the original path to the provided path,"

File: products/rules/src/content/transform/create-url-rewrite-rule.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    In the screenshot above, note that the **Expression Editor** area displays a text-only version of your expression. For more information on the Expression Editor, see [Edit rule expressions](https://developers.cloudflare.com/firewall/cf-dashboard/expression-preview-editor) in the Firewall documentation.
+
+1. Specify the value to match. If the value is an enumeration, then the **Value** control will be a drop-down list. Otherwise, it will be a text input. In this example the value _United Kingdom_ is set using the **Country** drop-down list.
+
+    ![Entering a field value in the Expression Builder](../static/transform/expression-value.png)
+
+1. Add any other filters to the rule expression. The example defines an additional filter for the _URI Path_, which must be equal to `/welcome.html`.
+
+1. To define the action for your URL Rewrite Rule, select one of the available options displayed as radio buttons, and then a value from the drop-down list, depending on the action.
```
Comment (kodster28) [inline products/rules/src/content/transform/create-url-rewrite-rule.md]: Remove the comma after "buttons"

File: products/rules/src/content/transform/create-url-rewrite-rule.md (hunk 7 above)
Comment (kodster28) [reply products/rules/src/content/transform/create-url-rewrite-rule.md]: Also, maybe shorten the start to "the _Static_ action..."

File: products/rules/src/content/transform/index.md [hunk 9]
```diff
@@ -5,37 +5,74 @@ order: 2
 
 # About Transform Rules
 
-Transform Rules allow you to make adjustments to incoming requests at the edge before they go through any other Cloudflare products. 
+Transform Rules allow you to make adjustments to incoming requests at the edge before they go through any other Cloudflare products.
 
-You can perform the following actions:
+There are two types of Transform Rules that perform different actions:
```
Comment (kodster28) [inline products/rules/src/content/transform/index.md]: think you could remove "that perform different actions"

File: products/rules/src/content/transform/index.md [hunk 10]
```diff
@@ -5,37 +5,74 @@ order: 2
...
+  <Button type="primary" href="/transform/create-header-modification-rule">Create a Header Modification Rule</Button>
+  <Button type="secondary" href="/transform/use-cases">Examples</Button>
+</ButtonGroup>
 
-## Rewrites and redirects 
+---
+
+## URL Rewrite Rules
+
+A URL Rewrite Rule can perform a **static rewrite** or a **dynamic rewrite**. Static rewrites replace a given part of a request URL (path or query string) with a static string. Dynamic rewrites allow you to support more advanced scenarios where you use an [expression](https://developers.cloudflare.com/firewall/cf-firewall-rules/fields-and-expressions) to define the resulting path or query string.
```
Comment (kodster28) [inline products/rules/src/content/transform/index.md]: "Dynamic rewrites support more advanced..."

File: products/rules/src/content/transform/index.md [hunk 11]
```diff
@@ -5,37 +5,74 @@ order: 2
...
 
 You can manipulate the URL of a request through different operations, namely through rewrites and redirects. 
 
 A **rewrite** is a server-side operation that occurs before a web server has fully processed a request. A rewrite is not visible to website visitors, since the URL displayed in the browser does not change. 
 
 A **redirect** is a client-side operation that occurs after the web server has loaded the initial URL. In this case, a website visitor can see the URL changing when the redirect occurs.
-You can configure Transform Rules in the dashboard to perform rewrites at the edge, without reaching your web server. 
 
-<Aside type='note' header='Note'>
+You can configure URL Rewrite Rules in the dashboard to perform rewrites at the edge, without reaching your web server. URL Rewrite Rules run in the `http_request_transform` Phase of Cloudflare's request handling workflow. For more information on Phases, check the [Ruleset Engine](https://developers.cloudflare.com/firewall/cf-rulesets) documentation.
```
Comment (kodster28) [inline products/rules/src/content/transform/index.md]: Remove the comma after "edge"?

File: products/rules/src/content/transform/index.md [hunk 12]
```diff
@@ -5,37 +5,74 @@ order: 2
...
+<Aside type='warning' label='Important'>
 
-![Transform Rules tab](../static/transform/overview.png)
+You cannot modify or remove HTTP headers whose name starts with `cf-` or `x-cf-` except for the `cf-connecting-ip` HTTP header, which you can remove.
+
+**Beta limitation:** Currently, you can only set the value of an HTTP Header using string values. You cannot use integer, IP, or boolean values such as `cf.bot_management.score`, `ip.src`, or `cf.bot_management.verified_bot`.
+
+</Aside>
+
+HTTP Header Modification Rules run in the `http_request_late_transform` Phase of Cloudflare's request handling workflow. For more information on Phases, see [Phases](https://developers.cloudflare.com/firewall/cf-rulesets#phases) in the Ruleset Engine documentation.
```
Comment (kodster28) [inline products/rules/src/content/transform/index.md:71]: Not sure I have a solution, but it feels weird to have two separate notes w/in the same warning block. But also don't think you should have two warnings.

Maybe it'd be better as a subsection named "Limitations"?

File: products/rules/src/content/transform/index.md [hunk 13]
```diff
@@ -45,20 +82,25 @@ This table outlines the Transform Rules features available with each customer pl
...
-Regular expression support | No   | No  | Yes      | Yes
+Feature                               | Free | Pro | Business | Enterprise
+--------------------------------------|------|-----|----------|-----------
+Active URL Rewrite Rules              | 2    | 5   | 10       | 20
+Active HTTP Header Modification Rules | 2    | 5   | 10       | 20
+Regular expression support            | No   | No  | Yes      | Yes
 
 </TableWrap>
 
+**Note:** During Beta, each plan has separate quotas for URL Rewrite Rules and HTTP Header Modification Rules. After Beta, these quotas will be unified in a single Transform Rules quota per plan.
```
Comment (kodster28) [inline products/rules/src/content/transform/index.md]: Should this be in an `Aside` for consistency?

File: products/rules/src/content/transform/create-header-modification-rule.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
+---
+title: Create a Header Modification Rule
+order: 2
+---
+
+# Create an HTTP Header Modification Rule
```
Comment (kodster28) [inline products/rules/src/content/transform/create-header-modification-rule.md:6]: Would it make sense to link out to the examples page from the top of this one?

File: products/rules/src/content/transform/create-url-rewrite-rule.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
+---
+order: 1
+---
+
+# Create a URL Rewrite Rule
```
Comment (kodster28) [inline products/rules/src/content/transform/create-url-rewrite-rule.md:5]: Would it make sense to link out to the examples page from the top of this one?

File: products/rules/src/content/transform/use-cases.md [hunk 16]
```diff
@@ -4,13 +4,27 @@ order: 22
...
+The following use cases illustrate how to perform URL rewrites with Transform Rules:
+
+* [Rewrite path of welcome page for visitors of specific countries](#rewrite-path-of-welcome-page-for-visitors-of-specific-countries)
+* [Rewrite URL query string of blog visitors](#rewrite-url-query-string-of-blog-visitors)
+* [Rewrite path of archived blog posts](#rewrite-path-of-archived-blog-posts)
+* [Rewrite path of moved section of a website](#rewrite-path-of-moved-section-of-a-website)
+* [Rewrite blog archive URLs to support a new URL format](#rewrite-blog-archive-urls-to-support-a-new-url-format)
 
-If you want to have a welcome page in two languages, create two Transform Rules with a static rewrite of the path component:
+### Rewrite path of welcome page for visitors of specific countries
```
Comment (kodster28) [inline products/rules/src/content/transform/use-cases.md]: Visitors **in** specific countries?

**Review by kodster28 — APPROVED** [review]

File: products/rules/src/content/transform/index.md (hunk 12 above)
Comment (pedrosousa) [reply products/rules/src/content/transform/index.md:71]: Used a couple of bullets in the Aside. The Beta limitation should disappear soon.

File: products/rules/src/content/transform/index.md (hunk 13 above)
Comment (pedrosousa) [reply products/rules/src/content/transform/index.md]: Updated.

File: products/rules/src/content/transform/create-header-modification-rule.md (hunk 14 above)
Comment (pedrosousa) [reply products/rules/src/content/transform/create-header-modification-rule.md:6]: Added new intro paragraph with link to examples.

File: products/rules/src/content/transform/create-url-rewrite-rule.md (hunk 15 above)
Comment (pedrosousa) [reply products/rules/src/content/transform/create-url-rewrite-rule.md:5]: Added new intro paragraph with link to examples.

**Pull request merged**